src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -184,11 +135,6 @@ public new CSharpParseOptions WithDocumentationMode(DocumentationMode documentat
                 return this;
             }
 
-            if (!documentationMode.IsValid())
-            {
-                throw new ArgumentOutOfRangeException(nameof(documentationMode));
-            }
-
             return new CSharpParseOptions(this) { DocumentationMode = documentationMode };
```

Comment:
Same issue with DocumentationMode. Since it is an enum specified programmatically, it should be diagnosed as an `ArgumentException` of some sort, not a diagnostic. See the [Compiler Design Notes](https://microsoft.sharepoint.com/teams/managedlanguages/_layouts/OneNote.aspx?id=%2Fteams%2Fmanagedlanguages%2Ffiles%2FTeam%20Notebook%2FRoslyn%20Team%20Notebook%20-%20New&wd=target%28Compilers%2FCompiler%20Design.one%7CD0427C1F-43BD-44AA-A337-20E46E56BB4C%2FRE%3A%20diagnostics%20vs.%20exceptions%7C3394BDFB-E7C9-40B6-B1C9-A8B87DD46ED7%2F%29) for guidance.